client/client.go

```diff
@@ -1069,17 +1069,23 @@ func (c *Client) updateNodeStatus() error {
 
 // updateAllocStatus is used to update the status of an allocation
 func (c *Client) updateAllocStatus(alloc *structs.Allocation) {
-	// Only send the fields that are updatable by the client.
+	select {
+	case c.allocUpdates <- alloc:
+	case <-c.shutdownCh:
+	}
+}
+
+// stripAllocation strips the allocation of fields which can be re-assembled in
+// the server.
+func (c *Client) stripAllocation(alloc *structs.Allocation) *structs.Allocation {
```

Comment:
Okay so `allocSync` seems like the wrong place to be adding that logic. That method is used just to batch updates. Can you move the logic to `updateAllocStatus` and revert back the stripAllocation changes because you will have the info you need there.